### PR TITLE
#168781679 User Login

### PR DESCRIPTION
**Pull request by alainmateso**
#### What does this PR do?
Add user login functionality.

#### Description of Task to be completed?
- creates an endpoint for a user to login
- gives user a token upon successful signin
- provides tests for login functionality

#### How should this be manually tested?
- Clone the repository from [here](https://github.com/andela/caret-bn-backend.git)
- Browse to the repo directory
- navigate to ft-user-login-168781679 branch
- Run `npm install`
- run `sequelize db:migrate`
- run `sequelize db:seed:all`
- run `npm run dev`
- In Postman Provide email and password in the body
- send a `POST` request to `http://localhost:3000/api/v1/users/login`

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#168781679
](https://www.pivotaltracker.com/story/show/168781679)

#### Screenshots (if appropriate)
Screen Shot 2019-10-10 at 20 09 32
Screen Shot 2019-10-10 at 20 10 01
Screen Shot 2019-10-10 at 20 10 22

#### Questions:
N/A

### CODE REVIEW

**Review by ngireric123 — APPROVED** [review]

File: .env.example [hunk 1]
```diff
@@ -28,3 +28,5 @@ APPLICATION_PORT=
 
 #JWT TOKEN
 JWT_SECRET=supersecretjwtkey
+# JWT SECRET
```
Comment (nignanthomas) [inline .env.example]: You don't need these lines 31-32.
Get rid of them

**Review by nignanthomas — CHANGES_REQUESTED** [review]
Great job @alainmateso so far
Just a few changes though
* Remove that unncessary variable in `.env.example`
* Update the PR body to the new `login` endpoint url `/api/v1/users/login`
* Update the screenshots too

**Review by nignanthomas — APPROVED** [review]

**Review by Cheza-Dzabala — APPROVED** [review]
Consider refactoring your controller

File: src/controllers/userController.js [hunk 2]
```diff
@@ -25,4 +27,27 @@ export default class UserController {
       });
     });
   }
+
+  static async signIn(req, res) {
+    const { email, password } = req.body;
+    const user = await models.users.findOne({ where: { email } });
```
Comment (nakiwuge) [inline src/controllers/userController.js:33]: Add a new line here

File: src/controllers/userController.js [hunk 3]
```diff
@@ -25,4 +27,27 @@ export default class UserController {
...
+    if (!user) {
+      return responseError(res, 400, strings.users.error.LOGIN_FAILURE);
+    }
+    const match = bcrypt.compareSync(password, user.password);
```
Comment (zellymiriam) [inline src/controllers/userController.js]: Add a new line here

File: src/controllers/userController.js [hunk 4]
```diff
@@ -25,4 +27,27 @@ export default class UserController {
       });
     });
   }
+
+  static async signIn(req, res) {
+    const { email, password } = req.body;
+    const user = await models.users.findOne({ where: { email } });
+    if (!user) {
+      return responseError(res, 400, strings.users.error.LOGIN_FAILURE);
+    }
```
Comment (nakiwuge) [inline src/controllers/userController.js:37]: Add a new line here

File: src/tests/loginTest.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
+import chai from 'chai';
+import chaiHttp from 'chai-http';
+import { describe, it } from 'mocha';
+import app from '../index';
+import strings from '../utils/stringsUtil';
+import mockData from './mockData/mockData';
+
```
Comment (nakiwuge) [inline src/tests/loginTest.js:7]: remove this line

**Review by nakiwuge — CHANGES_REQUESTED** [review]
Good job @alainmateso . The feature works as expected just a few changes are required to improve readability.

**Review by nakiwuge — APPROVED** [review]

**Pull request merged**